From f89ac505652bd5e5d9f2517685429b0da379c442 Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Fri, 18 Apr 2014 18:13:34 +0100
Subject: [PATCH 4/4] MythMusic General Settings: Fix Coverity ID 1196561
 Unchecked return value

Check the queries to truncate the music tables executed OK
---
 .../mythmusic/mythmusic/generalsettings.cpp        |   31 +++++++++++++++-----
 1 file changed, 23 insertions(+), 8 deletions(-)

diff --git a/mythplugins/mythmusic/mythmusic/generalsettings.cpp b/mythplugins/mythmusic/mythmusic/generalsettings.cpp
index 82a473b..ea0221d 100644
--- a/mythplugins/mythmusic/mythmusic/generalsettings.cpp
+++ b/mythplugins/mythmusic/mythmusic/generalsettings.cpp
@@ -125,21 +125,36 @@ void GeneralSettings::slotDoResetDB(bool ok)
         MSqlQuery query(MSqlQuery::InitCon());
 
         query.prepare("TRUNCATE music_albumart");
-        query.exec();
+        if (!query.exec())
+            MythDB::DBError("resetting music_albumart table", query);
+
         query.prepare("TRUNCATE music_albums");
-        query.exec();
+        if (!query.exec())
+            MythDB::DBError("resetting music_albums table", query);
+
         query.prepare("TRUNCATE music_artists");
-        query.exec();
+        if (!query.exec())
+            MythDB::DBError("resetting music_artists table", query);
+
         query.prepare("TRUNCATE music_directories");
-        query.exec();
+        if (!query.exec())
+            MythDB::DBError("resetting music_directories table", query);
+
         query.prepare("TRUNCATE music_genres");
-        query.exec();
+        if (!query.exec())
+            MythDB::DBError("resetting music_genres table", query);
+
         query.prepare("TRUNCATE music_playlists");
-        query.exec();
+        if (!query.exec())
+            MythDB::DBError("resetting music_playlists table", query);
+
         query.prepare("TRUNCATE music_songs");
-        query.exec();
+        if (!query.exec())
+            MythDB::DBError("resetting music_songs table", query);
+
         query.prepare("TRUNCATE music_stats");
-        query.exec();
+        if (!query.exec())
+            MythDB::DBError("resetting music_stats table", query);
 
         gMusicData->reloadMusic();
 
-- 
1.7.10.2

